include/sys/lock.h

```diff
@@ -1,27 +1,25 @@
 #ifndef _SYS_LOCK_H_
 #define _SYS_LOCK_H_
 
-/*! \enum Kind of lock. */
-typedef enum {
-  /*!\var LK_SLEEP
-   * \brief Type of sleeping lock.
-   *
-   * When a thread acquires sleeping lock that is owned by another thread it
-   * will be suspended and put on a sleep queue. */
-  LK_SLEEP = 0,
-  /*!\var LK_SPIN
-   * \brief Type of spinning lock.
-   *
-   * When a thread acquires spinning lock interrupts will be disabled.
-   */
-  LK_SPIN = 1,
-  /*!\var MTX_RECURSE
-   * \brief Type of recursive lock.
-   *
-   * The owner may lock it multiple times, but must release it as many times as
-   * she acquired it. */
-  LK_RECURSE = 2,
-} lock_type_t;
```

Comment:
IIRC There's one important advantage of having enums in C that may not be obvious. `gdb` understands values of enum types and print them symbolically. So instead of `3` you would get `LK_SPIN | LK_RECURSE`.

Please correct me if I'm wrong.

Reply:
You're right, it's a significant advantage of using enums over ints. It seems that using enums as flags isn't as bad as I thought it was.